### DISCUSSION TITLE
How can I increase the headersTimeout with Node.js's fetch?

### DISCUSSION THREAD

**Discussion by ImRodry** [nodejs/undici#1989] [Q&A] [answered]
I saw a PR that got merged recently into the latest 18.x release that supposedly increased the headers and body timeouts to 300s to match Chrome, however I'm still getting timed out in 30s. Is there a way to force the timeout to be whatever value I want?

**Comment by mcollina** [comment]
What Node version are you using?

**Reply by ImRodry** [reply]
The latest 18.14.2

**Reply by ImRodry** [reply]
So @mcollina do you know a fix? This is quite important to me

**Reply by metcoder95** [reply]
Are you using `fetch`?

**Reply by ImRodry** [reply]
Yes

**Reply by metcoder95** [reply]
Can you provide a 
[Minimum Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Reply by ImRodry** [reply]
Not really, it’s literally just any fetch call. Regardless I only want to know how to change that setting

**Reply by metcoder95** [reply, marked as answer]
Sadly we cannot help you that much without a 
[Minimum Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).

On a big overview, you cannot do that with fetch, as the `spec` does not support that configuration. One thing you can do, if wanting to use `fetch` in Node, is to pass a custom dispatcher that has all the timeout settings configured already.

e.g.
```js
import { fetch, Agent } from 'undici'

const res = await fetch('https://example.com', {
  // Mocks are also supported
  dispatcher: new Agent({
    keepAliveTimeout: 10,
    keepAliveMaxTimeout: 10
  })
})
const json = await res.json()
console.log(json)
```

Reference - https://undici.nodejs.org/#/?id=undicifetchinput-init-promise

**Reply by ImRodry** [reply]
Thanks, that’s all I wanted

**Comment by rhrn** [comment]
```js
import { Agent } from 'undici';

globalThis[Symbol.for('undici.globalDispatcher.1')] = new Agent({
  headersTimeout: 0,
  bodyTimeout: 0,
});
```
